specification/signalr/resource-manager/Microsoft.SignalR/preview/2018-03-01-preview/signalr.json

```diff
@@ -0,0 +1,1034 @@
+                    {
+                        "$ref": "#/parameters/ResourceGroupParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceNameParameter"
+                    }
+                ],
+                "responses": {
+                    "201": {
+                        "description": "Created and an async operation is excuting in background to make the new key to take effect. The response contains new keys and a Location header to query the async operation result.",
+                        "schema": {
+                            "$ref": "#/definitions/SignalRKeysResponse"
+                        }
+                    },
+                    "404": {
+                        "description": "Failed. The response indicates SignalR service doesn't exist."
+                    },
+                    "409": {
+                        "description": "Failed. The response describes the prerequisites for regeneration is not met."
+                    }
+                },
+                "x-ms-long-running-operation": true,
+                "x-ms-examples": {
+                    "RegenerateKey": {
+                        "$ref": "./examples/RegenerateKey.json"
+                    }
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SignalR/SignalR/{resourceName}": {
+            "get": {
+                "tags": [
+                    "SignalR"
+                ],
+                "description": "Get the SignalR service and its properties.",
+                "operationId": "SignalR_Get",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceNameParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Success. The response describe the corresponding SingalR service.",
+                        "schema": {
+                            "$ref": "#/definitions/SignalRResourceResponse"
+                        }
+                    }
+                },
+                "x-ms-examples": {
+                    "Get": {
+                        "$ref": "./examples/Get.json"
+                    }
+                }
+            },
+            "put": {
+                "tags": [
+                    "SignalR"
+                ],
+                "description": "Create a new SignalR service and update an exiting SignalR service.",
+                "operationId": "SignalR_Create",
```

Comment:
Should be `SignalR_CreateOrUpdate` if this allows you to update an existing resource